src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py

```diff
@@ -186,7 +208,14 @@ def Delete(self, cmd, client, resource_group_name, cluster_name, name, cluster_t
         user_confirmation_factory(cmd, yes)
 
     def Update(self, cmd, resource_group_name, cluster_name, auto_upgrade_minor_version, release_train, version, configuration_settings,
-               configuration_protected_settings, yes=False):
+               configuration_protected_settings, original_extension, yes=False):
+        input_configuration_settings = copy.deepcopy(configuration_settings)
+        input_configuration_protected_settings = copy.deepcopy(configuration_protected_settings)
+        # configuration_settings and configuration_protected_settings can be none, so need to set them to empty dict
+        if configuration_settings is None:
+            configuration_settings = {}
+        if configuration_protected_settings is None:
```

Comment:
This will always override the configuration_protected_settings and the configuration_settings in the PATCH call. Is this what you want? This may not be consistent with what the customer expects. i.e. If they don't make any updates to the configurationProtectedSettings, then the PATCH request that is generated shouldn't change them

Reply:
We overwrite in and keep the copy of input, if no changes are made in update flow, we will respect input None in 1.2.1 or empty dict in 1.2.0 and prior version.